Disable WebGPU Swift for Intel Macs

mwyrzykowski opened the pull request:
#### 48b742ce2b42925d7d10126d19ad7081b7692f3a
<pre>
Disable WebGPU Swift for Intel Macs
<a href="https://bugs.webkit.org/show_bug.cgi?id=300485">https://bugs.webkit.org/show_bug.cgi?id=300485</a>
<a href="https://rdar.apple.com/162340797">rdar://162340797</a>

Reviewed by Tadeu Zagallo.

Intel triggers OpenGLES verifier warnings, so disable
this config on WebGPU Swift for now.

* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebGPU/WebGPU/Buffer.mm:
(wgpuBufferCopy):
* Source/WebGPU/WebGPU/WebGPU.h:

Canonical link: <a href="https://commits.webkit.org/301339@main">https://commits.webkit.org/301339@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/44e900ae985d0c6e8ab1eba10f701927895bdc3a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125649 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45311 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36061 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132510 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77533 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/f1a62505-7553-42ed-a0de-d046ca8d7d97) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45998 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53873 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95724 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63840 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/a5669f1e-3141-4f04-a696-ae9c60bd8a01) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128597 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36771 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112363 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/76218 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/4b2ed6c3-a89b-4786-8b45-8ccf562c13c2) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35672 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75982 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106549 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30762 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/135182 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52444 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40207 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/104193 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52890 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108574 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103920 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26466 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49278 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27590 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/49667 "Hash 44e900ae for PR 52102 does not build (failure)") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52339 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/58141 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51687 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55040 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53383 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski commented on Source/WebGPU/WebGPU/Buffer.mm:
```diff
@@ -853,9 +853,14 @@ WGPUBufferUsageFlags wgpuBufferGetUsage(WGPUBuffer buffer)
 }
 
-#if ENABLE(WEBGPU_SWIFT)
 void wgpuBufferCopy(WGPUBuffer buffer, std::span<const uint8_t> data, size_t offset)
```
despite `ENABLE(WEBGPU_SWIFT)` being `0` for iosmac config, the verifiers would still complain that `wgpuBufferCopy(WGPUBuffer buffer, std::span<const uint8_t> data, size_t offset)` was an exported symbol without an implementation which didn't really make sense to me but I just unconditionally export it

tadeuzagallo replied:
that's very strange... I don't understand how that is possible, but if this fixes it then it's ok for now.

mwyrzykowski replied:
I'm going to see if disabling Catalyst support allows us to revert this change

mwyrzykowski commented on Source/WebGPU/Configurations/WebGPU.xcconfig:
```diff
@@ -95,8 +95,9 @@ ENABLE_WEBGPU_SWIFT[sdk=macosx*26*] = ENABLE_WEBGPU_SWIFT;
+ENABLE_WEBGPU_SWIFT[sdk=maccatalyst*] = ;
 ENABLE_WEBGPU_SWIFT[sdk=iphonesimulator26*] = ;
-ENABLE_WEBGPU_SWIFT[sdk=iphonesimulator26*.internal] = ENABLE_WEBGPU_SWIFT;
+ENABLE_WEBGPU_SWIFT[sdk=iphonesimulator26*.internal] = ;
```
WebGPU is not supported by the simulator, we don't need to worry about the simulator using Swift even on internal builds

rr-codes commented on Source/WebGPU/Configurations/WebGPU.xcconfig:
```diff
@@ -95,8 +95,9 @@ ENABLE_WEBGPU_SWIFT[sdk=macosx*26*] = ENABLE_WEBGPU_SWIFT;
+ENABLE_WEBGPU_SWIFT[sdk=maccatalyst*] = ;
 ENABLE_WEBGPU_SWIFT[sdk=iphonesimulator26*] = ;
-ENABLE_WEBGPU_SWIFT[sdk=iphonesimulator26*.internal] = ENABLE_WEBGPU_SWIFT;
+ENABLE_WEBGPU_SWIFT[sdk=iphonesimulator26*.internal] = ;
```
isn’t this redundant with the line above?

mwyrzykowski replied:
Yes :), will remove

tadeuzagallo commented on Source/WebGPU/Configurations/WebGPU.xcconfig:
```diff
@@ -95,8 +95,8 @@ ENABLE_WEBGPU_SWIFT[sdk=macosx*26*] = ENABLE_WEBGPU_SWIFT;
 ENABLE_WEBGPU_SWIFT[sdk=iphoneos26*] = ENABLE_WEBGPU_SWIFT;
 // FIXME: Support WebGPU swift in iOS simulator builds once the underlying
 // modules issue is fixed (https://bugs.webkit.org/show_bug.cgi?id=300146).
+ENABLE_WEBGPU_SWIFT[sdk=maccatalyst*] = ;
```
isn't this redundant? the flag is off by default and none of the regular expressions that enable it should match catalyst

mwyrzykowski replied:
oh maybe I only need lines 137 and 138 - checking right now

mwyrzykowski replied:
I see the original iosmac failure when I remove that line but I don't understand why

mwyrzykowski committed:
Disable WebGPU Swift for Intel Macs
https://bugs.webkit.org/show_bug.cgi?id=300485
rdar://162340797

Reviewed by Tadeu Zagallo.

Intel triggers OpenGLES verifier warnings, so disable
this config on WebGPU Swift for now.

* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebGPU/WebGPU/Buffer.mm:
(wgpuBufferCopy):
* Source/WebGPU/WebGPU/WebGPU.h:

Canonical link: https://commits.webkit.org/301339@main
